### ISSUE TITLE
List parameters

### ISSUE THREAD

**Issue by NeoDashboard** [CityOfZion/neo-mamba#253]
Hello,
I was wondering if there is a reason behind the choice of implementation of list parameters using `APPEND`/`DUP`/`NEWARRAY0` opcodes as It seems to not be necessary?

**Comment by ixje** [comment]
You mean this? 
https://github.com/CityOfZion/neo-mamba/blob/d7e2c025182166e32715516ad6af4b6591dcc4bd/neo3/vm.py#L333-L339

I'm open to suggestions on how you think we can improve it

**Comment by ixje** [comment]
Can you share an example how you're building your array?

**Comment by NeoDashboard** [comment]
I was using this on my side previously
```
class ScriptBuilderWithListSupport(vm.ScriptBuilder):  # type: ignore
    def emit_push_list(self, items: list):
        for item in reversed(items):
            if type(item) is list:
                self.emit_push_list(item)
                continue
            self.emit_push(item)
        self.emit_push(len(items))
        self.emit(vm.OpCode.PACK)

    def emit_dynamic_call_with_args(self, script_hash, operation: str, args) -> None:
        self.emit_push_list(args)
        self.emit_push(0xF)  # CallFlags.ALL
        self.emit_push(operation)
        self.emit_push(script_hash.to_array())
        self.emit_syscall(vm._syscall_name_to_int("System.Contract.Call"))

```

**Comment by NeoDashboard** [comment]
This?
```
    def emit_push_list(self, items: list):
        for item in reversed(items):
            if type(item) is list:
                self.emit_push_list(item)
                continue
            self.emit_push(item)
        self.emit_push(len(items))
        self.emit(vm.OpCode.PACK)
```

**Comment by NeoDashboard** [comment]
Something like this? https://github.com/CityOfZion/neo-mamba/pull/255

**Comment by Devel484** [comment]
> > It's been a while so I'm going to let it sink in and try to remember why I took this approach. If I can't come up with a good reason then we can change it to use `PACK` (should also be cheaper)
> 
> I can't think of anything at the moment, so open to accept a PR if you'd like

The previous implementation with DUP and APPEND was used to invert the order of the emitted items . The solution from @NeoDashboard emits the items directly in the correct (inversed) order and is cheaper.

**Comment by ixje** [comment]
Fixed in #256
